test/integration/mocks/mockchain.ts

```diff
@@ -26,12 +27,12 @@ export default class MockChain extends Chain {
     const blocks: any[] = []
     for (let number = 0; number < this.height; number++) {
       blocks.push(
-        new Block({
-          header: {
-            number: number + 1,
-            difficulty: 1,
+        Block.fromBlockData({
+          header: BlockHeader.fromHeaderData({
+            number: new BN(number + 1),
+            difficulty: new BN(1),
             parentHash: number ? blocks[number - 1].hash() : (this.genesis as any).hash,
-          },
+          }),
```

Comment:
Is this internal `BlockHeader.fromHeaderData()` necessary? I would have thought that the plain dict constructor should work here as well? 🤔

Reply:
Mmm, good question, not certain. Will take another look at this.